app/javascript/components/pages/FateNew.js

```diff
@@ -15,68 +14,127 @@ class FateNew extends Component{
-                  </Label>
-                </FormGroup>
-                <FormGroup check>
-                  <Label check>
-                    <Input type="radio" name="is_public"
-                     onChange={this.handleChange} value={false}/>private
-                  </Label>
-                </FormGroup>
-                <Button name="go-back" onClick={this.handleSubmit} className="button-style">
-                Go Back
-              </Button>
-              <Button name="submit" onClick={this.handleSubmit} className="button-style">
+  render() {
+    return (
+      <>
+        <h1>This is the FateNew page</h1>
+        <Form>
+          <FormGroup>
+            <br />
+            <Label id="Category">Category</Label>
+            <Input
+              type="text"
+              name="category"
+              onChange={this.handleChange}
+              value={this.state.form.category}
+            />
+          </FormGroup>
+          <Button
+            name="fetch-images"
+            onClick={this.fetchImages}
+            className="button-style"
+          >
+            get images
+          </Button>
+          <FormGroup>
+            <br />
+            <Label id="Url">Url</Label>
+            <Input
+              type="text"
+              name="url"
+              onChange={this.handleChange}
+              value={this.state.form.url}
+            />
+          </FormGroup>
+          <FormGroup>
+            <br />
+            <Label id="NewTable">Description</Label>
+            <Input
+              type="text"
+              name="description"
+              onChange={this.handleChange}
+              value={this.state.form.description}
+            />
+          </FormGroup>
+          <FormGroup check>
+            <Label check>
+              <Input
+                type="radio"
+                name="is_public"
+                onChange={this.handleChange}
+                value={true}
+              />
+              public
+            </Label>
+          </FormGroup>
+          <FormGroup check>
+            <Label check>
+              <Input
+                type="radio"
+                name="is_public"
+                onChange={this.handleChange}
+                value={false}
+              />
+              private
+            </Label>
+          </FormGroup>
+          <Button
+            name="go-back"
+            onClick={this.handleSubmit}
+            className="button-style"
+          >
+            Go Back
+          </Button>
+          <Button
+            name="submit"
+            onClick={this.handleSubmit}
+            className="button-style"
+          >
             submit
-              </Button>
-              </Form>
-
-              {this.state.submitted && <Redirect to="/fateindex" />}
-            </>
-
-        )
-    }
+          </Button>
+        </Form>
+        {this.state.submitted && <Redirect to="/fateindex" />}
+        {this.state.urls.map((url, keyID) => {
```

Comment:
what if there's no `this.state.urls`?

Reply:
would you be able to provide an example of how we dont get that? Or are you reffering to if like the search field doesnt work?

Reply:
i mean if you try to map and that property does not exist it will blow everything up... so you should do the optional chaining or the check `this.state.urls && <suff>`

Reply:
wouldn't this.state.urls?.map work in this situation? just asking if urls exists pretty much right?